templates/typescript/samples/sample-assistant/src/index.ts

```
@@ -156,34 +168,40 @@ try {
     const userProfileStateAccesor: StatePropertyAccessor<IUserProfileState> = userState.createProperty<IUserProfileState>('IUserProfileState');
-    const onboardingDialog: OnboardingDialog = new OnboardingDialog(userProfileStateAccesor, botServices , localeTemplateEngine, telemetryClient);
+    const onboardingDialog: OnboardingDialog = new OnboardingDialog(userProfileStateAccesor, botServices, localeTemplateEngine, telemetryClient);
     const switchSkillDialog: SwitchSkillDialog = new SwitchSkillDialog(conversationState);
     const previousResponseAccesor: StatePropertyAccessor<Partial<Activity>[]> =
-    userState.createProperty<Partial<Activity>[]>('Activity');
+        userState.createProperty<Partial<Activity>[]>('Activity');
 
-    // Register skill dialogs
-    const skillDialogs: SkillDialog[] = skills.map((skill: ISkillManifest): SkillDialog => {
-        const authDialog: MultiProviderAuthDialog|undefined = buildAuthDialog(skill, botSettings, appCredentials);
-        const credentials: MicrosoftAppCredentialsEx = new MicrosoftAppCredentialsEx(
-            botSettings.microsoftAppId || '',
-            botSettings.microsoftAppPassword || '',
-            skill.msaAppId);
 
-        return new SkillDialog(skill, credentials, telemetryClient, skillContextAccessor, authDialog);
-    });
-    
+    let skillHttpClient: SkillHttpClient = new SkillHttpClient(
+        credentianProvider,
+        new SkillConversationIdFactory(storage),
+        undefined
+    );
+
+    let skillDialogs: SkillDialog[] = [];
+    if (botSettings.skills !== undefined && botSettings.skills.length > 0) {
+        if (botSettings.skillHostEndpoint === undefined) {
+            throw new Error("'skillHostEndpoint' is not in the configuration");
+        }
+        
+        skillDialogs = botSettings.skills.map((skill: IEnhancedBotFrameworkSkill): SkillDialog => {
```

This should be added in an `else` statement.. https://github.com/microsoft/botframework-solutions/blob/10ae4d2f0fa595e9d1b1f5d5ba11f15fd3160e9a/templates/csharp/VA/VA/Startup.cs#L149

---

The else is already implicit since if `botSettings.skillHostEndpoint === undefined` is true, it will throw an error instead of continuing.

---

Yes, totally agree.. but we are not following the parity completely 😢